build_projects/dotnet-cli-build/CompileTargets.cs

```diff
@@ -157,7 +158,22 @@ private static void RemovePdbsFromDir(string directory)
             FS.RmFilesInDirRecursive(directory, "*.pdb");
         }
 
-        private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli dotnet, string rootOutputDirectory)
+        private static BuildTargetResult CompileCliSdkAndGenerateNuGetPackagesArchive(
```

Comment:
Did that. Also addressed @eerhardt comment about using stage1 to generate stage2's archive. Doing that too, which simplified the code a bit.